File: server/models/vendor_details.js
const mongoose = require("mongoose");
const { complaints } = require("./complaints");
const { reviews } = require("./reviews");

const vendorDetails = mongoose.Schema({
    vendorId: {
        type: String,
        trim: true,
        unique: true
    },
    name: {
        required: true,
        type: String,
        trim: true,
    },
    dob: {
        required: true,
        type: String
    },
    gender: {
        required: true,
        type: String
    },
    address: {
        type: String,
        default: "",
    },
    password: {
        required: true,
        type: String,
    },
    phone: {
        type: String,
        required: true,
        trim: true,
        validate: {
            validator: function(v) {
                return /^[0-9]{10}/.test(v);
            },
            message: '{VALUE} is not a valid 10 digit phone number!'
        }
    },
    aadharNo: {
        type: String,
        required: true,
        trim: true,
        validate: {
            validator: function(v) {
                return /^[0-9]{12}/.test(v);
            },
            message: '{VALUE} is not a valid Aadhar number!'
        }
    },
    panCardNo: {
        type: String,
        required: true,
        trim: true,
        validate: {
            validator: function(v) {
                return /[A-Z]{5}[0-9]{4}[A-Z]{1}/.test(v);
            },
            message: '{VALUE} is not a valid Pancard number!'
        }
    },
    isPassport: {
        type: Boolean,
        required: true
    },
    isElectionid: {
        type: Boolean,
        required: true
    },
    isMcgmLicense: {
        type: Boolean,
        required: true
    },
    aadharcardImageUrl: {
        type: String,
        required: true
    },
    pancardImageUrl: {
        type: String,
        required: true
    },
    shopLocationAddress: {
        type: String,
        required: true
    },
    shopLocationLat: {
        type: Number,
        required: true
    },
    shopLocationLong: {
        type: Number,
        required: true
    },
    vendingZoneIdApplied: {
        type: String,
        required: true
    },
    shopCity: {
        type: String,
        required: true
    },
    creditScore: {
        type: Number,
        required: true,
        default: 0
    },
    vendorImageurl: {
        type: String,
        default: "https://avatars.githubusercontent.com/u/84977709?v=4"
    },
    isApproved: {
        type: String,
        default: "pending"
    },
    complaintsList: {
        type: [complaints]
    },
    reviewList: {
        type : [reviews]
    },
    weeklyBazzarList: [{
      "bazzarId" : {
        type: String,  
      },
      "bazzarName" : {
        type: String,
       
      },
      "status" : {
        type: String,
        default : "pending"
      } 
}]  ,
    schemeList :[{
      "schemeId" : {
        type: String,
        unique : true, 
        
      },
      "status" : {
        type: String,
        default : "pending"
      }
    }], 
    vendorCategory : {
      type : String , 
      required : true 
    }, 
    shopName : {
        type : String ,
        required : true ,
        trim : true
    }, 

    currentLat : {
        type : Number , 
    

    }, 
    currentLong : {
        type : Number , 
    }, 
    inTime : {
        type : [String] ,

    }, 
    outTime : {
        type : [String] ,
    },
    inOrOut : {
        type : Boolean ,
        default : true 
    }
})

const Vendor = mongoose.model("Vendor", vendorDetails);
module.exports = Vendor;

File: App/vendo/lib/models/vendorDetailsModel/vendor_details.dart
import 'dart:convert';

import 'package:json_annotation/json_annotation.dart';
import 'package:vendo/models/vendorComplainsModel/vendor_complaints.dart';

part 'vendor_details.g.dart';

@JsonSerializable()
class VendorModel {
  String vendorId;
  String name;
  String dob;
  String gender;
  String address;
  String password;
  String phone;
  String aadharNo;
  String panCardNo;
  bool isPassport;
  bool isElectionid;
  bool isMcgmLicense;
  String aadharcardImageUrl;
  String pancardImageUrl;
  String shopLocationAddress;
  String token;
  String vendingZoneIdApplied;
  List<dynamic> reviewList;
  List<VendorComplaintModel> complaintsList;
  List<dynamic> weeklyBazzarList;
  double creditScore;
  String vendorImageurl;
  String isApproved;
  double shopLocationLat;
  double shopLocationLong;
  String vendorCategory;
  String shopCity;
  String shopName;
  List<dynamic> schemeList;
  double currentLat;
  double currentLong;
  List<dynamic> inTime;
  List<dynamic> outTime;
  bool inOrOut;

  VendorModel({
    required this.vendorId,
    required this.name,
    required this.dob,
    required this.gender,
    required this.address,
    required this.password,
    required this.phone,
    required this.aadharNo,
    required this.panCardNo,
    required this.isPassport,
    required this.isElectionid,
    required this.isMcgmLicense,
    required this.aadharcardImageUrl,
    required this.pancardImageUrl,
    required this.shopLocationAddress,
    required this.token,
    required this.vendingZoneIdApplied,
    required this.reviewList,
    required this.complaintsList,
    required this.weeklyBazzarList,
    required this.creditScore,
    required this.vendorImageurl,
    required this.isApproved,
    required this.shopLocationLat,
    required this.shopLocationLong,
    required this.vendorCategory,
    required this.shopCity,
    required this.shopName,
    required this.schemeList,
    required this.currentLat,
    required this.currentLong,
    required this.inOrOut,
    required this.inTime,
    required this.outTime,
  });

  factory VendorModel.fromJson(Map<String, dynamic> json) =>
      _$VendorModelFromJson(json);

  Map<String, dynamic> toJson() => _$VendorModelToJson(this);
}
